src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -519,21 +512,29 @@ public byte[] Http2AltSvcOriginUri
             Socket? socket = null;
             SslStream? sslStream = null;
             TransportContext? transportContext = null;
+            Http2Connection[]? lastHttp2Connections = _http2Connections;
 
             // Serialize creation attempt
             await _http2ConnectionCreateLock.WaitAsync(cancellationToken).ConfigureAwait(false);
             try
             {
-                if (_http2Connection != null)
+                Http2Connection[]? currentHttp2Connections = _http2Connections;
+                if (EnableMultipleHttp2Connections)
                 {
-                    // Someone beat us to it
-
-                    if (NetEventSource.Log.IsEnabled())
+                    if (_http2Connections != null)
                     {
-                        Trace("Using existing HTTP2 connection.");
+                        //Connection expiration will be checked here as well.
+                        Http2Connection? existingConnection = GetExistingHttp2Connection();
```

Comment:
This is a nit, but I don't think we need a separate `existingConnection` variable here. We already have `http2Connection` defined above and we're consistently using this variable throughout the function, so it seems like we should just use it here as well.

Reply:
Fixed.